src/psyclone/psyGen.py

```diff
@@ -3083,6 +3083,129 @@ def scalars(self):
             "Arguments.scalars must be implemented in sub-class")
 
 
+class DataAccess(object):
+    '''A helper class to handle dependencies due to overlapping data
+    accesses between fields
+
+    '''
+
+    def __init__(self, arg):
+        '''store the source argument and its parent call. Also set any
+        internal coverage data.
+
+        :param arg: the argument that we are concerned with
+        :type arg: :py:class:`psyclone.psyGen.Argument`
+
+        '''
+        self._arg = arg
```

Comment:
Please add comments describing what `self._arg` and `self._call` are

Reply:
OK, although I'm just repeating myself.